js/widgets/temperament.js

```diff
@@ -2169,11 +2157,68 @@ class TemperamentWidget {
                 setTimeout(() => {
-                    __playLoop(i);
+                    if (this.closed) {
+                        this.closed = !this.closed;
+                        return;
+                    } else if (!this._playing) {
+                        cell.innerHTML =
+                            '&nbsp;&nbsp;<img src="header-icons/' +
+                            "play-button.svg" +
+                            '" title="' +
+                            _("Play") +
+                            '" alt="' +
+                            _("Play") +
+                            '" height="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" width="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" vertical-align="middle" align-content="center">&nbsp;&nbsp;';
+                        if (i !== -1) {
+                            if (this.circleIsVisible == false && docById("wheelDiv4") == null) {
+                                this.notesCircle.navItems[i - 1].fillAttr = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].sliceHoverAttr.fill = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].slicePathAttr.fill = "#c8C8C8";
+                                this.notesCircle.navItems[i - 1].sliceSelectedAttr.fill = "#c8C8C8";
+                                if (i == 11) {
+                                    //on completion of a full circle and on hitting '0' note in clockwise direction
+                                    this.notesCircle.navItems[0].fillAttr = "#c8C8C8";
+                                    this.notesCircle.navItems[0].sliceHoverAttr.fill = "#c8C8C8";
+                                    this.notesCircle.navItems[0].slicePathAttr.fill = "#c8C8C8";
+                                    this.notesCircle.navItems[0].sliceSelectedAttr.fill = "#c8C8C8";
+                                } else if (i < 11) {
+                                    //in case of counter-clockwise direction, i.e., when this.playbackForward = false
+                                    this.notesCircle.navItems[i + 1].fillAttr = "#c8C8C8";
+                                    this.notesCircle.navItems[i + 1].sliceHoverAttr.fill =
+                                        "#c8C8C8";
+                                    this.notesCircle.navItems[i + 1].slicePathAttr.fill = "#c8C8C8";
+                                    this.notesCircle.navItems[i + 1].sliceSelectedAttr.fill =
+                                        "#c8C8C8";
+                                }
+                                this.notesCircle.refreshWheel();
+                            } else if (
+                                this.circleIsVisible == true &&
+                                docById("wheelDiv4") == null
+                            ) {
+                                j = i - 1;
```

Comment:
omit variable

Reply:
Can you please elaborate?

Reply:
this variable can be omitted, it's a bad practice to declare temp variables if they are used only once.
i,j => naming should be meaningful for other people who read your code.

Reply:
Done.